src/components/editor/AppController.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { browserHistory } from 'react-router';
+import FalcorController from "lib/falcor/FalcorController";
+import { setAppReady } from "lib/falcor/falcorUtils";
+
+export default class AppController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      defaultValue: this.handleDefaultValue.bind(this)()
+    })
+    
+  }
+  static getFalcorPath() {
+    return;
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    setAppReady();
+  }
+
+  handleDefaultValue() {
+    let path = this.props.location.pathname;
+    if (path === '/') {
+      return '/';
+    }
+    else if (path[2] === 'r') {
+      return 'articles';
+    }
+    else {
+      return 'authors';
+    }
+  }
+
+  handleChange(event) {
```

Comment:
should have more specific name. handleModeChange or something.

Though again I feel like this should be done in react-router and not here.